libsignal-service/src/push_service.rs

```diff
@@ -271,10 +278,25 @@ pub enum ServiceError {
     #[error("Service request timed out: {reason}")]
     Timeout { reason: String },
 
+    #[error("invalid URL: {0}")]
+    InvalidUrl(#[from] url::ParseError),
+
     #[error("Error sending request: {reason}")]
     SendError { reason: String },
+
+    #[error("Error decoding response: {reason}")]
+    ResponseError { reason: String },
+
     #[error("Error decoding JSON response: {reason}")]
     JsonDecodeError { reason: String },
+    #[error("Error decoding protobuf frame: {0}")]
+    ProtobufDecodeError(#[from] prost::DecodeError),
+    #[error("Error encoding protobuf frame: {0}")]
+    ProtobufEncodeError(#[from] prost::EncodeError),
```

Comment:
This shouldn't be necessary if you replace all the encode(&mut Vec) stuff with `with_capacity` and `.expect()`. Encoding is infallible except when the vector doesn't grow. See https://github.com/danburkert/prost/pull/378 :-)